Updating reset link colors

clrux opened the pull request:
This quick fix addresses the HEX colors in our `_reset.scss` file for links. A recent merged piece of work simplified link styles allowing them to rely more on inheritance, rather than on a bunch of deeply specified (and often overwritten and conflicting styles). However this caused visual discrepancies on the course dashboard, mainly with the link colors. I discovered the reset file was using an incorrect color blue.

- [x] @talbs

talbs commented on lms/static/sass/base/_reset.scss:
```diff
@@ -9,9 +9,9 @@ html { font-size: 100%; -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 10
-a { color: #00e; }
-a:visited { color: #551a8b; }
-a:hover, a:focus { color: #06e; }
+a { color: #005584; } // from the Pattern Library
```
Any way we can pipe through Sass vars for these? It'd be great to get those in as early as possible.

clrux replied:
We'd have to move this file under the variables in the compile. Not sure what kind of nuclear bomb that would drop. Want me to try?

talbs replied:
Hmm, nahh - I don't want that can of worms opened up at the risk of not getting this fix in. If you have time and want to try as a separate PR off of this (or once this merges), that sounds like a good thing.

talbs replied:
NIT: also, mind updating your comments to point to the exact URL of the sass where these colors are defined

talbs commented:
@clrux, when I fired this up locally I'm noticing some weird things:

![screen shot 2015-09-09 at 9 21 23 am](https://cloud.githubusercontent.com/assets/163763/9762866/36ef9938-56d4-11e5-9f3f-1a90ae49d790.png)

* The hover/focus state's color property is what you're setting in ``_reset.scss`` but there's little to no visual contrast. Do the values you've added really sync up with https://github.com/edx/ux-pattern-library/blob/master/pattern-library/sass/utilities/_variables.scss#L370?
* The specificity and overriding that's happening (see the inspector in my screengrab) is really hairy. Where are those other base ``a`` or ``a:state`` rules coming from? Can we make sure the reset colors aren't causing these issues/can overcome these?

clrux commented:
@talbs A few notes:

* LMS previously had our focus/hover color as `$pink`, but I've changed this to sync up with our library, which uses a lighter (`base`) blue rather than the pink.
* The mess of other styles are coming from a host of other link rules from a number of other stylesheets. I searched for `"a:visited"` and cleaned up the found lines, but because everything is compiled into a singe stylesheet it's hard to tell where exactly the rules are coming from.

talbs commented:
> The mess of other styles are coming from a host of other link rules from a number of other stylesheets. I searched for "a:visited" and cleaned up the found lines, but because everything is compiled into a singe stylesheet it's hard to tell where exactly the rules are coming from.

Thanks for digging. I'm fine with any follow-up work on certain states to be done in other planned PRs. For now I just care that the Dashboard course title links have some sort of hover state to note that they can be clicked on (that isn't purple :stuck_out_tongue: ).

talbs commented on lms/static/sass/base/_variables.scss:
```diff
@@ -463,7 +463,7 @@ $border-color-l4: $m-gray-l4;
 $link-color: $blue;
 $link-color-d1: $blue;
-$link-hover: $pink;
+$link-hover: $blue-l1; // Was formerly $pink, but is synced up with our Pattern Library http://ux.edx.org/elements/colors/
```
No need to say that this was formerly ``$pink`` in your comment. Just mirror your other uxpl comments.

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -104,7 +104,6 @@ $learn-more-horizontal-position: calc(50% - 100px); // calculate the left positi
           margin: ($baseline*0.25) 0 ($baseline*1.75) 0;
           padding: 0 ($baseline*0.75);
           height: $course-title-height;
-          color: $link-color;
```
Why was this taken out? Is it needed or does the reset work now cover this? Have you tested it locally?

talbs replied:
I just confirmed locally that this causes a display issue.

![image](https://cloud.githubusercontent.com/assets/163763/9766135/65dc5220-56e5-11e5-9e01-8d76fee6164b.png)

Please make sure this works as it does on master now.

talbs commented:
@clrux, a few more small comments on this and then it should be good to merge.

talbs commented:
@clrux, also please revise your commits so that the message describes the part of the platform/app affected (e.g. ``LMS: updating hover colors for visited links``).

talbs commented:
:+1:
